{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/users/auth/tokens.py

```diff
@@ -34,11 +37,40 @@ def get_user_for_token(token, scope):
     except jwt.DecodeError:
         raise exc.NotAuthenticated("Invalid token")
 
-    model_cls = apps.get_model("users", "User")
+    model_cls = get_user_model()
 
     try:
         user = model_cls.objects.get(pk=data["user_%s_id" % (scope)])
     except (model_cls.DoesNotExist, KeyError):
         raise exc.NotAuthenticated("Invalid token")
     else:
         return user
+
+
+def get_token_for_password_reset(user):
+    return "{}::{}".format(encode_uuid_to_base64(user.pk), PasswordResetTokenGenerator().make_token(user))
+
+
+def get_user_for_password_reset_token(token):
+    default_error_messages = {
+        'invalid_token': 'Invalid token or the token has expired',
+        'user_not_found': 'No user exists for given token'
+    }
+    try:
+        uuid_value, reset_token = token.split("::")
+    except ValueError:
+        raise exc.RequestValidationError(default_error_messages['invalid_token'])
+
+    user_id = decode_uuid_from_base64(uuid_value)
+    if not user_id:
+        raise exc.RequestValidationError(default_error_messages['invalid_token'])
+
+    user = get_user_model().objects.filter(id=user_id).first()
+
+    if not PasswordResetTokenGenerator().check_token(user, reset_token):
+        raise exc.RequestValidationError(default_error_messages['invalid_token'])
+
+    if not user:
```

Comment:
It should be moved above `PasswordResetTokenGenerator` check as if user is None than that check will fail first and `not user` condition will never occur.